src/write_gcm.c

```diff
@@ -108,6 +113,10 @@ static const char json_content_type_header[] = "Content-Type: application/json";
 //==============================================================================
 //==============================================================================
 
+static _Bool wg_end_to_end_test_mode() {
+  return atoi(global_option_get("ReadThreads")) == -1;
```

Comment:
This seems a little fragile for the future to depend on to know that this is the mode we're in.